File: src/desktop/legacy/main_window/main_widget/learn_tab/lesson_configs.py
from data.constants import LETTER


LESSON_CONFIGS = {
    "Lesson1": {
        "question_format": "pictograph",
        "answer_format": "button",
        "quiz_description": "pictograph_to_letter",
        "question_prompt": "Choose the letter for:",
    },
    "Lesson2": {
        "question_format": LETTER,
        "answer_format": "pictograph",
        "quiz_description": "letter_to_pictograph",
        "question_prompt": "Choose the pictograph for:",
    },
    "Lesson3": {
        "question_format": "pictograph",
        "answer_format": "pictograph",
        "quiz_description": "valid_next_pictograph",
        "question_prompt": "Which pictograph can follow?",
    },
}

File: legacy_web_app/src/lib/state/stores/learn/lesson_configs.ts
export type QuestionFormat = 'text' | 'pictograph';
export type AnswerFormat = 'button' | 'pictograph';

export interface LessonConfig {
	id: string;
	title: string;
	description: string;
	questionFormat: QuestionFormat;
	answerFormat: AnswerFormat;
	prompt: string;
	options?: {
		numOptions?: number; // Number of answer options to show
		randomizeOptions?: boolean; // Whether to randomize the order of options
	};
}

export const lessonConfigs: LessonConfig[] = [
	{
		id: 'letter_to_pictograph',
		title: 'Letter to Pictograph',
		description: 'Learn to identify the pictograph that represents a given letter',
		questionFormat: 'text',
		answerFormat: 'pictograph',
		prompt: 'Choose the pictograph for:',
		options: {
			numOptions: 4,
			randomizeOptions: true
		}
	},
	{
		id: 'pictograph_to_letter',
		title: 'Pictograph to Letter',
		description: 'Learn to identify the letter represented by a pictograph',
		questionFormat: 'pictograph',
		answerFormat: 'button',
		prompt: 'Choose the letter for:',
		options: {
			numOptions: 5,
			randomizeOptions: true
		}
	},
	{
		id: 'turns',
		title: 'Turn Recognition',
		description: 'Learn to identify clockwise and counterclockwise turns in pictographs',
		questionFormat: 'pictograph',
		answerFormat: 'button',
		prompt: 'Identify the turn pattern:',
		options: {
			numOptions: 3,
			randomizeOptions: true
		}
	},
	{
		id: 'positions',
		title: 'Position Recognition',
		description: 'Learn to recognize the positions represented in pictographs',
		questionFormat: 'pictograph',
		answerFormat: 'button',
		prompt: 'Identify the position:',
		options: {
			numOptions: 4,
			randomizeOptions: true
		}
	}
];
